src/DI/Definitions/FactoryDefinition.php

```diff
@@ -231,6 +231,13 @@ private function completeParameters(Nette\DI\Resolver $resolver): void
 		$interface = $this->getType();
 		$method = new \ReflectionMethod($interface, self::METHOD_CREATE);
 
+		if (!$this->resultDefinition instanceof ServiceDefinition) {
+			throw new ServiceCreationException(
+				'Result definition must be type of ServiceDefinition, '
+				. "type '" . get_class($this->resultDefinition) . "' given."
```

Comment:
This cannot work. `setResultDefinition()` allows `Definition`. I would personally allow only `ServiceDefinition`, but I guess it's an intention and most common return type is defined in annotation just because generics support is not good enough yet.
cc @dg

Reply:
Yes, true. I reverted old state.